apps/back/src/DevTools/PHPStan/RouteSecurityChecker.php

```diff
@@ -0,0 +1,184 @@
+                    "Pass the 'subject' argument to the \$this->denyAccessUnlessGranted() call.
+                    If you are sure that this route's protection should only on user's permissions, add a ".ThisRouteDoesntNeedAVoter::class." attribute.",
+                );
+            }
+        }else{
+            $isGrantedAttributeInstance = $isGrantedAttribute->newInstance();
+            \assert($isGrantedAttributeInstance instanceof IsGranted);
+            if ($isGrantedAttributeInstance->subject === null) {
+                return $this->buildError(
+                    sprintf('🛑🔓 SECURITY: Route %s::%s is insufficiently protected !', $className, $functionName),
+                    "Pass the 'subject' argument to the 'IsGranted' attribute.
+                    If you are sure that this route's protection should only on user's permissions, add a ".ThisRouteDoesntNeedAVoter::class.' attribute.',
+                );
+            }
+        }
+
+        return [];
+    }
+
+    /** @param array<ReflectionAttribute<object>> $attributes */
+    private function functionAttributesContain(string $attributeClass, array $attributes): bool
+    {
+        return \count($this->getAttributes($attributeClass, $attributes)) > 0;
+    }
+
+    /**
+     * @param array<ReflectionAttribute<object>> $attributes
+     *
+     * @return ReflectionAttribute<object>|null
+     */
+    private function getAttribute(string $attributeClass, array $attributes): ReflectionAttribute|null
+    {
+        $attributes = $this->getAttributes($attributeClass, $attributes);
+
+        return \count($attributes) > 0 ? $attributes[0] : null;
+    }
+
+    /**
+     * @param array<ReflectionAttribute<object>> $attributes
+     *
+     * @return array<ReflectionAttribute<object>>
+     */
+    private function getAttributes(string $attributeClass, array $attributes): array
+    {
+        return array_values(array_filter(
+            $attributes,
+            static fn (ReflectionAttribute $attr) => $attr->getName() === $attributeClass
+        ));
+    }
+
+    /**
+     * @return array<RuleError>
+     */
+    private function buildError(string $message, string $tip): array
+    {
+        return [
+            RuleErrorBuilder::message($message . "\n")
+                ->tip($tip)
+                ->build(),
+        ];
+    }
+
+    private function isDenyAccessUnlessGrantedCalledInRouteFunction(InClassMethodNode $node, bool $requireSubject): bool
+    {
+        $visitor = new class ($requireSubject) extends NodeVisitorAbstract {
+            private bool $isSecurityCheckFunctionCalled = false;
+
+            public function __construct(private readonly bool $requireSubject)
+            {
+            }
+
+            public function enterNode(Node $node): int|null
+            {
+                // phpcs:disable SlevomatCodingStandard.ControlStructures.EarlyExit.EarlyExitNotUsed
```

Comment:
it should work with this CI rule with the current form

Reply:
Indeed :)